setup.py

```diff
@@ -38,6 +41,7 @@
 }
 POSTFIX_SHA256[('cygwin', 'x86_64')] = POSTFIX_SHA256[('win32', 'AMD64')]
 POSTFIX_SHA256[('darwin', 'arm64')] = POSTFIX_SHA256[('darwin', 'x86_64')]
+POSTFIX_SHA256[('linux', 'armv7l')] = POSTFIX_SHA256[('linux', 'armv6hf')]
```

Comment:
matching armv7l to armv6hf does not seem correct

Reply:
It works on my raspi, which identifies with armv7l. The binaries are compatible, and there is no native armv7l version of it.

Reply:
i believe the binary compatibility is unidirectional, i.e. arm7l can run all arm6hf binaries, but not necessarily the other way around. is that correct?

Reply:
I think this is correct

Reply:
@asottile how do you feel about having arch fallbacks? e.g. for this case, if `armv7l` exists then use that, but otherwise fall back to `armv6hf`.

Reply:
before we allow that, we should make sure that it actually is correct -- if so that's probably fine -- we're already doing a fallback for M1

Reply:
@blackliner are you able to point us to any official statement (e.g. documentation) from arm that can confirm the binary compatibility?

Reply:
not directly, but:

1. I ran `armv6hf` binaries on `armv7l` (both 32bit!) successfully in the past
2. http://single-boards.com/armv6-vs-armv7/ -->
```
ARMv7 is backward compatible with ARMv6, so binaries compiled for ARMv6 should also work on ARMv7
```

Reply:
so i think armv6hf means there's a hardware fpu. and armv7l means it's little endian.

endianness aside, i think float ops in a armv6hf executable may fail to run on an armv7l processor if the processor doesn't have a fpu. i'm not super certain about this, and it may be the case that all armv7l processors have an fpu?

in any case, let me think about this a bit. we may ultimately merge it for now, with the intention of reverting it we get any reports of breakage.